xbmc/cores/VideoPlayer/Interface/CaptionBlock.h

```diff
@@ -0,0 +1,67 @@
+/*
+ *  Copyright (C) 2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include <stdint.h>
+#include <vector>
+
+/*!
+ \brief Abstracts a Closed Caption data block
+ */
+class CCaptionBlock
+{
+public:
+  /*!
+  \brief Closed Caption block constructor
+    *
+    * \param[in] pts - the presentation timestamp of this closed caption
+    * \param[in] data - the data to store on the block
+    * \param[in] size - the size of the data to store
+  */
+  CCaptionBlock(double pts, uint8_t* data, int size) : m_pts(pts)
+  {
+    m_data.resize(size);
+    std::copy(data, data + size, m_data.data());
+  }
+
+  CCaptionBlock(const CCaptionBlock&) = delete;
+  CCaptionBlock& operator=(const CCaptionBlock&) = delete;
+  ~CCaptionBlock() = default;
+
+  /*!
+  \brief Get the size of the caption block
+    *
+    * \return the size of the closed caption block
+  */
+  unsigned long Size() const { return m_data.size(); }
+
+  /*!
+  \brief Get the presentation timestamp (pts) of the closed caption block
+    *
+    * \return the pts of the closed caption block
+  */
+  double GetPTS() const { return m_pts; }
+
+  /*!
+  \brief Get the data stored in the closed caption block
+    *
+    * \return the data stored in the closed caption block
+  */
+  std::vector<uint8_t> GetData() const { return m_data; }
```

Comment:
This could be an expensive operation, as it always copies `m_data`. What about returning a const reference to m_data?